tls/s2n_handshake_io.c

```diff
@@ -617,13 +666,7 @@ int s2n_conn_set_handshake_type(struct s2n_connection *conn)
     }
 
     if (IS_TLS13_HANDSHAKE(conn)) {
-        /* Use middlebox compatibility mode for TLS1.3 by default.
-         * For now, only disable it when QUIC support is enabled. */
-        if (!conn->config->quic_enabled) {
-            conn->handshake.handshake_type |= MIDDLEBOX_COMPAT;
-        }
-        conn->handshake.handshake_type |= FULL_HANDSHAKE;
-
+        GUARD_AS_POSIX(s2n_conn_set_tls13_handshake_type(conn));
         return 0;
     }
```

Comment:
We could separate the logic into two separate static functions and call them according to the version of TLS being set: 

```
int s2n_conn_set_handshake_type(struct s2n_connection *conn) { 

    if (IS_TLS13_HANDSHAKE(conn)) {
            GUARD_AS_POSIX(s2n_conn_set_tls13_handshake_type(conn));
    } else { 
           GUARD_AS_POSIX(s2n_conn_set_pre_tls13_handshake_type(conn));
    } 
}
```

Reply:
In the past, we've generally handled these situations by making a separate TLS1.3 method, but keeping the existing pre-TLS1.3 method as-is. Possibly because it makes the diff of what changed clearer?

Reply:
I see, the reason I was suggesting separating the code into different static methods was to makes it more readable and less prone to errors in logic flow.